app/views/documents/index/_results.html.erb

```diff
@@ -38,6 +38,10 @@
           <%= t "documents.index.search_results.state",
             state: t("user_facing_states.#{document.user_facing_state}.name")
           %>
+          <br/>
+          <%= t "documents.index.search_results.last_edited_by",
+            user: document.last_edited_by&.name || document.creator&.name || I18n.t("documents.index.search_results.unknown_creator")
```

Comment:
Do we have a N+1 query here? I think that for every document we'll end up doing a query to fetch the latest timeline entry, and then a query to fetch the user. For 50 results that would mean 100 extra database calls.

Reply:
That's a good point.  I've been swinging between this approach and just writing the user who last edited to the document.  This is making me think the latter is going to be more performant.  What do you think?

Reply:
One way would be to do `self.timeline_entries.includes(:user)`, which will prevent the 2nd query, but that still leaves the 50 for the entries.

Another approach depends on what the original user need for this feature is. I could imagine that it would be useful for users to see all of the people involved, rather than just the last, so you'd have something like this:

```
Publishers: Tijmen, Emma and Kevin
```

All of the participating people is easier to fetch in a single query with something like:

```rb
documents = # whatever we search for
User.where(id: TimelineEntry.where(document: documents).select(:user_id))
```

@kevindew what do you think?

Reply:
Update: above doesn't at all solve the problem, I don't know why I thought it would.

Reply:
Ok, this would work. The trick is to do the queries but do the filtering in Ruby:

```rb
Document
  .all
  .includes(timeline_entries: [:user])
  .each { |d| 
    puts d.timeline_entries
      .sort_by(&:created_at)
      .select { |t| t.entry_type.in?(%w(updated_content updated_tags)) }
      .first.user.name 
  }
```

Reply:
I'd imagine it's good to go down the route of explicitly storing who last edited the document on the document. Then you can get around the N+1 problem. 

I think if you're dealing with a list of documents you're not going to want to deal a list of x length of publishers.

I think we'll handle the full grouping through the history.

It'd probably useful to have in the filterable options selecting anyone who has ever edited a document but I think that could be achieved in a relatively performant way of doing an exists sub query on timeline entry.

Reply:
Given that we're already storing the creator on the document itself, it's consistent if we store the editor on the document too, although it's a shame as we could potentially use the history for both of these things. Are the user needs different in some way?